cranelift/jit/src/backend.rs

```diff
@@ -893,7 +893,7 @@ fn lookup_with_dlsym(name: &str) -> Option<*const u8> {
     use windows_sys::Win32::Foundation::HINSTANCE;
     use windows_sys::Win32::System::LibraryLoader;
 
-    const MSVCRT_DLL: &[u8] = b"msvcrt.dll\0";
+    const UCRTBASE: &[u8] = b"ucrtbase.dll\0";
```

Comment:
I'm slightly concerned about this change.

`msvcrt` does not define `ceilf`/`floorf`/`nearbyintf`/`truncf`, which are some of the LibCall's that we define. I cannot find the dll that defines those functions pre VS2015.

After VS2015 Microsoft created the Universal C Runtime which defines `ucrtbase.dll` that does contain all of those functions. However that only became standard with Windows 10 in 2015. Previous versions of Windows can have that dll, but only if they install Visual Studio 2015.

We cannot search `msvcrt` and fallback to `ucrtbase` since mixing the two runtimes will cause weird behaviour, probably at the worst time possible.

Does anyone know where we can find those functions? Or alternativley what are the minimum requirements that we have for cranelift and can we upgrade to UCRT?

Reply:
I'm told Windows 8.1 reaches end of life in January, which seems to me like a good enough reason for Cranelift to require Windows 10. But others may disagree.

Since we're looking at using the libm crate for #4517, I thought I'd check if we could get these functions from there, instead of worrying about Windows C runtime versions. It does have `ceilf`, `floorf`, and `truncf`, but unfortunately not `nearbyintf`, so I'm not sure if that helps with this question.

Reply:
@cfallin I would like to move forward with this PR since it's blocking some stuff on the fuzzer (getting function calls working).

Would @jameysharp 's reasoning that pretty soon all supported windows versions are going to have `ucrt` be sufficient? An alternative could be to keep using `msvcrt`, but we would need to disable some ops on the runtest suite, which is also not ideal.

We can also go with polyfill's as @jameysharp mentioned, I don't like this approach since we would probably keep needing to patch holes in msvcrt. In https://github.com/bjorn3/rustc_codegen_cranelift/pull/1254 we also found that `fabsf` is missing from msvcrt, so that's going to be another one that we need. With the bonus that we know that one is [definitely broken in libm](https://github.com/bytecodealliance/wasmtime/pull/4517#issuecomment-1195138197).

Reply:
Yeah, I think that's fine; Win10 as a minimum requirement here is probably an acceptable compromise, given the context.